src/icarus/userinterface.py

```diff
@@ -0,0 +1,170 @@
+from tkinter import Tk, Label, Entry, Button, Frame, Checkbutton, BooleanVar, StringVar
+from pathlib import Path
+
+class UserInterface:
+    """Provides a user interface using tkinter to allow the user to control aspects of the modelling process,
+        including selecting parameters to be perturbed and defining their values and how many validation datasets
+        to generate for each parameter.
+    """
```

Comment:
For this class level doctring I would keep it brief at least for the topline first sentence, if you want to document more of the functionality, you can start a paragraph below but it's usually best to start with a single line brief description.
```suggestion
    """User interface enabling users to control aspects of the modelling process.
    """
```